### PR TITLE
Feature/web smart sync all locator

### PR DESCRIPTION
**Pull request by GokulBothe**
Thank you for your contribution.
Before submitting this PR, please make sure:

- [ ] PR description and commit message should describe the changes done in this PR
- [ ] Verify the PR is pointing to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [ ] Latest Code from master or specific release branch is merged to your branch
- [ ] No unwanted\commented\junk code is included
- [ ] No new warning upon build solution
- [ ] Code Summary\Comments are added to my code which explains what my code is doing
- [ ] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [ ] Sanity Tests are successfully executed for New and Existing Functionality
- [ ] Verify that changes are compatible with all relevant browsers and platforms.
- [ ] After creating pull request there should not be any conflicts
- [ ] Resolve all Codacy comments
- [ ] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments
- [ ] Update the Help Library document to match any feature changes

## Summary by CodeRabbit

- **New Features**
  - Added a checkbox for "Try all supported locators" in the web element synchronization settings.

- **Bug Fixes**
  - Improved alignment of menu icons for better visual consistency.

- **Tests**
  - Introduced unit tests for enhanced validation of web element locator functionality in the Selenium driver.

- **Refactor**
  - Enhanced flexibility in the POM (Page Object Model) methods to support overrides.
  
- **Style**
  - Updated icon margins in the main menu for better alignment.

### CODE REVIEW

File: Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/Selenium/SeleniumDriver.cs [hunk 1]
```diff
@@ -2194,187 +2200,230 @@ private void ActPasswordHandler(ActPassword actPassword)
...
+                if (currentPOMElementInfo == null)
+                {
+                    throw new Exception("Unable to find details about the POM");
+                }
+                ElementLocator firstLocator=null;
+                if (act.UseAllLocators == true)
                 {
-                    POMExecutionUtils pomExcutionUtil = new(act, act.ElementLocateValue);
-                    if (pomExcutionUtil.GetCurrentPOM() == null)
+                    foreach (var l in currentPOMElementInfo.Locators)
```
Comment (Maheshkale447) [inline Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/Selenium/SeleniumDriver.cs:2228]: How is it trying for all available locators where it is returning first active supported locator from line 2234

**Review by Maheshkale447 — CHANGES_REQUESTED** [review]
Change target branch to master and review comments by coderabbit

File: Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/Selenium/SeleniumDriver.cs (hunk 1 above)
Comment (GokulBothe) [reply Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/Selenium/SeleniumDriver.cs]: changed code

File: Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/Selenium/SeleniumDriver.cs [hunk 2]
```diff
@@ -2194,203 +2200,290 @@ private void ActPasswordHandler(ActPassword actPassword)
...
+                eLocateBy.ByID => By.Id(locateValue),
+                eLocateBy.ByName => By.Name(locateValue),
+                eLocateBy.ByClassName => By.ClassName(locateValue),
+                eLocateBy.ByCSSSelector => By.CssSelector(locateValue),
+                eLocateBy.ByLinkText => By.LinkText(locateValue),
+                eLocateBy.ByTagName => By.TagName(locateValue),
+                _ => throw new Exception("Supported locator values include: ByXPath, ByID, ByName, ByClassName, ByCssSelector, ByLinkText, ByRelativeXpath and ByTagName."),
+            };
+            return elementLocator;
+        }
```
[the remark this answers is not in the document]
Comment (GokulBothe) [reply Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/Selenium/SeleniumDriver.cs:2278]: changes done

File: Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/Selenium/SeleniumDriver.cs [hunk 3]
```diff
@@ -104,6 +104,12 @@ public class SeleniumDriver : GingerWebDriver, IVirtualDriver, IWindowExplorer,
         String[] SeleniumUserArgs = null;
         DriverService driverService = null;
         private readonly List<string> HighlightStyleList = ["arguments[0].style.outline='3px dashed rgb(239, 183, 247)'", "arguments[0].style.backgroundColor='rgb(239, 183, 247)'", "arguments[0].style.border='3px dashed rgb(239, 183, 247)'"];
+        static List<ActWebSmartSync.eSyncOperation> operationsWithoutLocator =
+          [
+              ActWebSmartSync.eSyncOperation.AlertIsPresent,
+                ActWebSmartSync.eSyncOperation.PageHasBeenLoaded,
+                ActWebSmartSync.eSyncOperation.UrlMatches
+          ];
```
[the remark this answers is not in the document]
Comment (GokulBothe) [reply Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/Selenium/SeleniumDriver.cs:112]: this way is not working

**Pull request closed without merging**